src/Operations.jl

```diff
@@ -1045,7 +1045,7 @@ function develop(ctx::Context, pkgs::Vector{PackageSpec}, new_git::Vector{UUID};
     # resolve & apply package versions
     resolve_versions!(ctx, pkgs)
     update_manifest!(ctx, pkgs)
-    new_apply = download_source(ctx, pkgs; readonly=false)
+    new_apply = download_source(ctx, pkgs; readonly=true)
```

Comment:
Even though this is in `develop` we should install the dependencies as read-only.